### ISSUE TITLE
Please enable issues to be reflected on the mailing list

### ISSUE THREAD

**Issue by webpayments** [w3c/webpayments#9]
Issues are currently not reflected on the mailing list. Please enable that feature.

**Comment by msporny** [comment]
On 11/12/2015 01:23 PM, webpayments wrote:

> Issues are currently not reflected on the mailing list. Please
> enable that feature.

I got the email, so reflection of issues should work.

-- manu

## 

Manu Sporny (skype: msporny, twitter: manusporny, G+: +Manu Sporny)
Founder/CEO - Digital Bazaar, Inc.
blog: Web Payments: The Architect, the Sage, and the Moral Voice
https://manu.sporny.org/2015/payments-collaboration/

**Comment by shepazu** [comment]
Seems like a good idea.

**Comment by webpayments** [comment]
Ok, testing the mailing list version of this to see if it works.

**Comment by halindrome** [comment]
I don't think people who do not have github accounts can respond and have their responses captured.  But I am not sure.

**Comment by shepazu** [comment]
Can a non-account-holder post to Github?

**Comment by webpayments** [comment]
On 11/12/2015 01:34 PM, Shane McCarron wrote:

> I don't think people who do not have github accounts can respond and 
> have their responses captured. But I am not sure.

No, but getting a github user account isn't a terrible thing... we'll
have to try it out w/ people that don't have Github accounts.

Doug and I are testing this being reflected to a W3C email listserv. We
think the listserv is eating the responses, but are not quite sure yet.

-- manu

## 

Manu Sporny (skype: msporny, twitter: manusporny, G+: +Manu Sporny)
Founder/CEO - Digital Bazaar, Inc.
blog: Web Payments: The Architect, the Sage, and the Moral Voice
https://manu.sporny.org/2015/payments-collaboration/

**Comment by webpayments** [comment]
Testing another email address, that doesn't have a Github account.

**Comment by msporny** [comment]
Testing to see if commits show up on the mailing list.

**Comment by adrianhopebailie** [comment]
I'm not convinced that pushing back to GitHub from the mailing list is important.
Am I alone in thinking that?

i.e. We are tracking issues on GitHub. The mailing list is useful in tracking activity but if you want to comment on an issue or create a new issue you should do it on GitHub.

There are various advantages to this, not least of which is the ability for me to mention someone like @msporny and they get a direct email from GitHub to notify them.

**Comment by halindrome** [comment]
I think the concern is that some people in our community will be unable to
create github accounts because of security constraints in their
organizations.  There may be other concerns as well, but that was voiced in
Sapporo.

On Fri, Nov 13, 2015 at 6:03 AM, Adrian Hope-Bailie <
notifications@github.com> wrote:

> I'm not convinced that pushing back to GitHub from the mailing list is
> important.
> Am I alone in thinking that?
> 
> i.e. We are tracking issues on GitHub. The mailing list is useful in
> tracking activity but if you want to comment on an issue or create a new
> issue you should do it on GitHub.
> 
> There are various advantages to this, not least of which is the ability
> for me to mention someone like @msporny https://github.com/msporny and
> they get a direct email from GitHub to notify them.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/w3c/webpayments/issues/9#issuecomment-156413017.

## 

Shane McCarron
halindrome@gmail.com

**Comment by adrianhopebailie** [comment]
My recollection of the consensus in Sapporo was that there were some that said they would have restrictions against accessing GitHub but they would make a plan to work around these. On that basis we have chosen to use GitHub as our primary work tool.

If there are any in the group that have issues with using Github then let's hear from them and address those concerns directly, make a decision about our tooling and move on to doing great things :smile:

**Comment by halindrome** [comment]
Works for me.  I'm never one to borrow trouble. I have plenty of my own.

On Fri, Nov 13, 2015 at 7:01 AM, Adrian Hope-Bailie <
notifications@github.com> wrote:

> My recollection of the consensus in Sapporo was that there were some that
> said they would have restrictions against accessing GitHub but they would
> make a plan to work around these. On that basis we have chosen to use
> GitHub as our primary work tool.
> 
> If there are any in the group that have issues with using Github then
> let's hear from them and address those concerns directly, make a decision
> about our tooling and move on to doing great things [image: :smile:]
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/w3c/webpayments/issues/9#issuecomment-156426269.

## 

Shane McCarron
halindrome@gmail.com

**Comment by msporny** [comment]
There are four reasons to push back to the mailing list via email that I can think of:
1. To filter certain mails will require someone to write email processing software specific to Github responses. Not allowing responses from the mailing list would be more work - someone would have to sit down and write the mail filter software and maintain the bounce server.
2. It enables those that can't go out to Github to participate in the discussion while they're at work. When working on some of the collaboratively edited documents in the WPIG, I often got responses from folks that couldn't access the collaboration sites via their work computers. This resulted in their inability to participate.
3. It ensures that if someone "accidentally" replies via the mailing list that their comment is recorded in the appropriate place rather than being lost in the mailing list (or requiring the chairs to sync up all comments that were not put into Github to do so manually).
4. It doesn't force people to adopt a new workflow if they don't want to. Folks that are used to the old W3C mailing-list based workflow can continue to work like that. Folks that want to use the new Github workflow can do that too. The software takes care of sync'ing both workflows. If we can, we should work with people's workflows instead of forcing them into a new one. This is especially true if there is an easy way to do this (and given the solution we've put together, there is an easy way to do this).

I'd prefer to not delay deploying this solution any longer than we have. If there are concerns about the solution, let's iterate on the solution rather than delaying until something better to come along. I've already checked w/ W3C staff on this and at least Doug is ok with going forward with this solution (since it's better than the one W3C has).

**Comment by msporny** [comment]
To be clear, here's the process that I followed to set the issue mirroring up (we're waiting on approval (or a timeout at 1pm today) by Dominique at W3C):

Here's a process that doesn't necessarily need to involve pinging you (and enables WGs to manage their own repository subscriptions).
1. Setup an email address for the express purpose of reflecting github emails to a W3C mailing list. I setup w3c-webpayments-wg@digitalbazaar.com, but this alias should eventually be set to github-w3c-webpayments-wg@w3.org (or something similar). Set the reflection address to be your personal email address just for step #2. To do this, you need to have access to an email server.
2. Setup a new Github user for the Working Group with the email address from step #1. For the Web Payments WG, I setup this account: https://github.com/webpayments/
3. Verify the email address for the new user by checking your email and clicking on the Github email verification link.
4. Set the mailing list alias created in step #1 to the WGs mailing list.
5. As the user created in step #1, start "Watch"ing the repositories of interest to the WG, which can be any Github repo.

NOTE: W3C Staff contacts might have to watch the Admin/spam filters on the mailing list to approve the incoming mail, but the test email address we used went through w/o admin approval

Benefits of this approach:
1. People don't have to contact a single bottleneck at W3C (Dominique) to manage WG mailing list subscriptions. This takes work off of Dominique's plate and enables WGs to operate more independently.
2. People can respond to the emails and they're reflected on Github whether or not one has  a Github account.
3. Any arbitrary number of github repos can be subscribed to in this way, so as groups create new repos, management of subscriptions is as simple as the WG github user "Watch"ing each new repo. You don't need to keep contacting Dominique when a new repo is added.

Downsides to this approach:
1. Anyone that has access to the mailing list can respond to issues, even if they don't have a Github account (this could be used to spam particular issues). Response is to delete the spam comment. I can't see an upside to this attack other than annoying a group.
2. If someone sends a password reset code for the user, it'll go to the mailing list. This could be counteracted by filtering passcode reset emails or enabling two-factor auth and not allowing reset by email.

**Comment by msporny** [comment]
Hi all,

Reflecting Github issues to the Web Payments WG mailing list is now active. You can also respond directly to the email to have your comments reflected back on the Github issue tracker. I'll send another comment in a bit to explain how this is setup and how we should manage subscriptions to Github repositories and issues.

**Comment by webpayments** [comment]
*** NOTE: ***

    Your message was sent from an address which is not on the list
    of people who are authorized to post to this mailing list.
    Therefore, your message has been forwarded to the list maintainer
    for manual processing.

    If you do not see your message appear on the list or in the
    mailing list archives within a few business days, you may wish
    to contact the mailing list maintainer to investigate the delay.

    -- W3C Postmaster
       http://www.w3.org/Mail/

your message follows:
----------------------------------------------------------------------------


----==_mimepart_564630bd87620_71ee3fc34dde72b84028b6
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

Hi all,

Reflecting Github issues to the Web Payments WG mailing list is now active. You can also respond directly to the email to have your comments reflected back on the Github issue tracker. I'll send another comment in a bit to explain how this is setup and how we should manage subscriptions to Github repositories and issues.

---
Reply to this email directly or view it on GitHub:
https://github.com/w3c/webpayments/issues/9#issuecomment-156520486
----==_mimepart_564630bd87620_71ee3fc34dde72b84028b6
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>Hi all,</p>

<p>Reflecting Github issues to the Web Payments WG mailing list is now active. You can also respond directly to the email to have your comments reflected back on the Github issue tracker. I'll send another comment in a bit to explain how this is setup and how we should manage subscriptions to Github repositories and issues.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/w3c/webpayments/issues/9#issuecomment-156520486">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/APFrHA4Qggzq9BoS0ri2F_gJ0UKi9jHWks5pFig9gaJpZM4GhP95.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/w3c/webpayments/issues/9#issuecomment-156520486"></link>
  <meta itemprop="name" content="View Issue"></meta>
</div>
<meta itemprop="description" content="View this Issue on GitHub"></meta>
</div>

----==_mimepart_564630bd87620_71ee3fc34dde72b84028b6--

**Comment by msporny** [comment]
Hi all, 

Mailing list traffic for the Web Payments Working Group is going to start increasing in the next day or two. This is due to hooking up Github issue notifications to the public-payments-wg@w3.org mailing list. Here's how it works:
- All issue discussion related to Github repositories that we're watching will be reflected on the mailing list.
- If you want to respond to the issue, the best way to do so is to click on the link at the end of the message and respond via Github's issue tracker.
- If you do not have access to Github, just "Reply" to the email. Please don't "Reply All" as it'll result in two emails being sent out.

This approach enables those that want to work in Github to do so while ensuring that those that can't work via Github can still participate in the discussion. I will send a separate email outlining how the Chairs and W3C Staff Contacts should manage messages.

**Comment by webpayments** [comment]
*** NOTE: ***

    Your message was sent from an address which is not on the list
    of people who are authorized to post to this mailing list.
    Therefore, your message has been forwarded to the list maintainer
    for manual processing.

    If you do not see your message appear on the list or in the
    mailing list archives within a few business days, you may wish
    to contact the mailing list maintainer to investigate the delay.

    -- W3C Postmaster
       http://www.w3.org/Mail/

your message follows:
----------------------------------------------------------------------------


----==_mimepart_5647a210bb8bf_6d163fc5ef3db2c011130c0
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

Hi all, 

Mailing list traffic for the Web Payments Working Group is going to start increasing in the next day or two. This is due to hooking up Github issue notifications to the public-payments-wg@w3.org mailing list. Here's how it works:

* All issue discussion related to Github repositories that we're watching will be reflected on the mailing list.
* If you want to respond to the issue, the best way to do so is to click on the link at the end of the message and respond via Github's issue tracker.
* If you do not have access to Github, just "Reply" to the email. Please don't "Reply All" as it'll result in two emails being sent out.

This approach enables those that want to work in Github to do so while ensuring that those that can't work via Github can still participate in the discussion. I will send a separate email outlining how the Chairs and W3C Staff Contacts should manage messages.

---
Reply to this email directly or view it on GitHub:
https://github.com/w3c/webpayments/issues/9#issuecomment-156745954
----==_mimepart_5647a210bb8bf_6d163fc5ef3db2c011130c0
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>Hi all, </p>

<p>Mailing list traffic for the Web Payments Working Group is going to start increasing in the next day or two. This is due to hooking up Github issue notifications to the <a href="mailto:public-payments-wg@w3.org">public-payments-wg@w3.org</a> mailing list. Here's how it works:</p>

<ul>
<li>All issue discussion related to Github repositories that we're watching will be reflected on the mailing list.</li>
<li>If you want to respond to the issue, the best way to do so is to click on the link at the end of the message and respond via Github's issue tracker.</li>
<li>If you do not have access to Github, just "Reply" to the email. Please don't "Reply All" as it'll result in two emails being sent out.</li>
</ul>

<p>This approach enables those that want to work in Github to do so while ensuring that those that can't work via Github can still participate in the discussion. I will send a separate email outlining how the Chairs and W3C Staff Contacts should manage messages.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/w3c/webpayments/issues/9#issuecomment-156745954">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/APFrHH8486M4yM_A1mvTfaLWzOQwStMAks5pF5mQgaJpZM4GhP95.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/w3c/webpayments/issues/9#issuecomment-156745954"></link>
  <meta itemprop="name" content="View Issue"></meta>
</div>
<meta itemprop="description" content="View this Issue on GitHub"></meta>
</div>

----==_mimepart_5647a210bb8bf_6d163fc5ef3db2c011130c0--

**Comment by webpayments** [comment]
Hi Manu,

Not sure what list is intended but I”m getting a moderator action…

Ian

> On Nov 14, 2015, at 3:05 PM, Manu Sporny notifications@github.com wrote:
> 
> Hi all,
> 
> Mailing list traffic for the Web Payments Working Group is going to start increasing in the next day or two. This is due to hooking up Github issue notifications to the public-payments-wg@w3.org mailing list. Here's how it works:
> 
>   • All issue discussion related to Github repositories that we're watching will be reflected on the mailing list.
>   • If you want to respond to the issue, the best way to do so is to click on the link at the end of the message and respond via Github's issue tracker.
>   • If you do not have access to Github, just "Reply" to the email. Please don't "Reply All" as it'll result in two emails being sent out.
> This approach enables those that want to work in Github to do so while ensuring that those that can't work via Github can still participate in the discussion. I will send a separate email outlining how the Chairs and W3C Staff Contacts should manage messages.
> 
> —
> Reply to this email directly or view it on GitHub.

## 

Ian Jacobs ij@w3.org      http://www.w3.org/People/Jacobs
Tel:                       +1 718 260 9447

**Comment by adrianhopebailie** [comment]
Hi Manu,

To clarify what has been setup.
- There is now a bot Github user (https://github.com/webpayments/) with
  the email address github-w3c-webpayments-wg@w3.org
- That email account is subscribed to the WG mailing list and is also
  subscribed to all Github issue notifications on the repo:
  https://github.com/w3c/webpayments
- When that account receives mails from Github it forwards them to the
  mailing list.
- When the account receives replies from the mailing list it forwards
  them to Github.

The result (once all moderation issues have been resolved) will be that:
- If a new issue is logged or a comment is added to an existing issue
  then a mail will be sent to the mailing list
- Anyone replying to mails on the mailing list related to issues will
  see their replies posted against the Github issue.

We still recommend that members of the WG post their comments and issues
directly on Github if possible as this has the least chance of creating
SPAM, messing with formatting or sending out duplicate messages.

Are there any outstanding security concerns:
1. Is it possible to compromise the Github account of the bot?
2. Is it possible to post to Github "anonymously"?
3. Are we certain that non email sent to the members-only mailing list
   will end up on Github?

Thanks for making this work. If it's as seamless as it appears it should be
then I think it will be a great way to use Guthub and still accommodate
those that like to (or have to) use the mailing list.

Adrian

**Comment by webpayments** [comment]
*** NOTE: ***

    Your message was sent from an address which is not on the list
    of people who are authorized to post to this mailing list.
    Therefore, your message has been forwarded to the list maintainer
    for manual processing.

    If you do not see your message appear on the list or in the
    mailing list archives within a few business days, you may wish
    to contact the mailing list maintainer to investigate the delay.

    -- W3C Postmaster
       http://www.w3.org/Mail/

your message follows:
----------------------------------------------------------------------------


----==_mimepart_5649a6b63281c_65343f8b4515b2a096388
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: quoted-printable

Hi Manu,

To clarify what has been setup.


   - There is now a bot Github user (https://github.com/webpayments/) wit=
h
   the email address github-w3c-webpayments-wg@w3.org
   - That email account is subscribed to the WG mailing list and is also
   subscribed to all Github issue notifications on the repo:
   https://github.com/w3c/webpayments
   - When that account receives mails from Github it forwards them to the=

   mailing list.
   - When the account receives replies from the mailing list it forwards
   them to Github.

The result (once all moderation issues have been resolved) will be that:

   - If a new issue is logged or a comment is added to an existing issue
   then a mail will be sent to the mailing list
   - Anyone replying to mails on the mailing list related to issues will
   see their replies posted against the Github issue.

We still recommend that members of the WG post their comments and issues
directly on Github if possible as this has the least chance of creating
SPAM, messing with formatting or sending out duplicate messages.

Are there any outstanding security concerns:

   1. Is it possible to compromise the Github account of the bot?
   2. Is it possible to post to Github "anonymously"?
   3. Are we certain that non email sent to the members-only mailing list=

   will end up on Github?

Thanks for making this work. If it's as seamless as it appears it should =
be
then I think it will be a great way to use Guthub and still accommodate
those that like to (or have to) use the mailing list.

Adrian


On 15 November 2015 at 21:32, webpayments <notifications@github.com> wrot=
e:

> Hi Manu,
>
> Not sure what list is intended but I=E2=80=9Dm getting a moderator acti=
on=E2=80=A6
>
> Ian
>
> > On Nov 14, 2015, at 3:05 PM, Manu Sporny <notifications@github.com>
> wrote:
> >
> > Hi all,
> >
> > Mailing list traffic for the Web Payments Working Group is going to
> start increasing in the next day or two. This is due to hooking up Gith=
ub
> issue notifications to the public-payments-wg@w3.org mailing list. Here=
's
> how it works:
> >
> > =E2=80=A2 All issue discussion related to Github repositories that we=
're
> watching will be reflected on the mailing list.
> > =E2=80=A2 If you want to respond to the issue, the best way to do so =
is to click
> on the link at the end of the message and respond via Github's issue
> tracker.
> > =E2=80=A2 If you do not have access to Github, just "Reply" to the em=
ail. Please
> don't "Reply All" as it'll result in two emails being sent out.
> > This approach enables those that want to work in Github to do so whil=
e
> ensuring that those that can't work via Github can still participate in=
 the
> discussion. I will send a separate email outlining how the Chairs and W=
3C
> Staff Contacts should manage messages.
> >
> > =E2=80=94
> > Reply to this email directly or view it on GitHub.
> >
>
> --
> Ian Jacobs <ij@w3.org> http://www.w3.org/People/Jacobs
> Tel: +1 718 260 9447
>
>
>
>
> =E2=80=94
> Reply to this email directly or view it on GitHub
> <https://github.com/w3c/webpayments/issues/9#issuecomment-156845681>.
>


---
Reply to this email directly or view it on GitHub:
https://github.com/w3c/webpayments/issues/9#issuecomment-156973244=

----==_mimepart_5649a6b63281c_65343f8b4515b2a096388
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: quoted-printable

Hi Manu,<br>
<br>
To clarify what has been setup.<br>
<br>
<br>
   - There is now a bot Github user (https://github.com/webpayments/) wit=
h<br>
   the email address github-w3c-webpayments-wg@w3.org<br>
   - That email account is subscribed to the WG mailing list and is also<=
br>
   subscribed to all Github issue notifications on the repo:<br>
   https://github.com/w3c/webpayments<br>
   - When that account receives mails from Github it forwards them to the=
<br>
   mailing list.<br>
   - When the account receives replies from the mailing list it forwards<=
br>
   them to Github.<br>
<br>
The result (once all moderation issues have been resolved) will be that:<=
br>
<br>
   - If a new issue is logged or a comment is added to an existing issue<=
br>
   then a mail will be sent to the mailing list<br>
   - Anyone replying to mails on the mailing list related to issues will<=
br>
   see their replies posted against the Github issue.<br>
<br>
We still recommend that members of the WG post their comments and issues<=
br>
directly on Github if possible as this has the least chance of creating<b=
r>
SPAM, messing with formatting or sending out duplicate messages.<br>
<br>
Are there any outstanding security concerns:<br>
<br>
   1. Is it possible to compromise the Github account of the bot?<br>
   2. Is it possible to post to Github &quot;anonymously&quot;?<br>
   3. Are we certain that non email sent to the members-only mailing list=
<br>
   will end up on Github?<br>
<br>
Thanks for making this work. If it&#39;s as seamless as it appears it sho=
uld be<br>
then I think it will be a great way to use Guthub and still accommodate<b=
r>
those that like to (or have to) use the mailing list.<br>
<br>
Adrian<br>
<br>
<br>
On 15 November 2015 at 21:32, webpayments &lt;notifications@github.com&gt=
; wrote:<br>
<br>
&gt; Hi Manu,<br>
&gt;<br>
&gt; Not sure what list is intended but I=E2=80=9Dm getting a moderator a=
ction=E2=80=A6<br>
&gt;<br>
&gt; Ian<br>
&gt;<br>
&gt; &gt; On Nov 14, 2015, at 3:05 PM, Manu Sporny &lt;notifications@gith=
ub.com&gt;<br>
&gt; wrote:<br>
&gt; &gt;<br>
&gt; &gt; Hi all,<br>
&gt; &gt;<br>
&gt; &gt; Mailing list traffic for the Web Payments Working Group is goin=
g to<br>
&gt; start increasing in the next day or two. This is due to hooking up G=
ithub<br>
&gt; issue notifications to the public-payments-wg@w3.org mailing list. H=
ere&#39;s<br>
&gt; how it works:<br>
&gt; &gt;<br>
&gt; &gt; =E2=80=A2 All issue discussion related to Github repositories t=
hat we&#39;re<br>
&gt; watching will be reflected on the mailing list.<br>
&gt; &gt; =E2=80=A2 If you want to respond to the issue, the best way to =
do so is to click<br>
&gt; on the link at the end of the message and respond via Github&#39;s i=
ssue<br>
&gt; tracker.<br>
&gt; &gt; =E2=80=A2 If you do not have access to Github, just &quot;Reply=
&quot; to the email. Please<br>
&gt; don&#39;t &quot;Reply All&quot; as it&#39;ll result in two emails be=
ing sent out.<br>
&gt; &gt; This approach enables those that want to work in Github to do s=
o while<br>
&gt; ensuring that those that can&#39;t work via Github can still partici=
pate in the<br>
&gt; discussion. I will send a separate email outlining how the Chairs an=
d W3C<br>
&gt; Staff Contacts should manage messages.<br>
&gt; &gt;<br>
&gt; &gt; =E2=80=94<br>
&gt; &gt; Reply to this email directly or view it on GitHub.<br>
&gt; &gt;<br>
&gt;<br>
&gt; --<br>
&gt; Ian Jacobs &lt;ij@w3.org&gt; http://www.w3.org/People/Jacobs<br>
&gt; Tel: +1 718 260 9447<br>
&gt;<br>
&gt;<br>
&gt;<br>
&gt;<br>
&gt; =E2=80=94<br>
&gt; Reply to this email directly or view it on GitHub<br>
&gt; &lt;https://github.com/w3c/webpayments/issues/9#issuecomment-1568456=
81&gt;.<br>
&gt;<br>


<p style=3D"font-size:small;-webkit-text-size-adjust:none;color:#666;">&m=
dash;<br>Reply to this email directly or <a href=3D"https://github.com/w3=
c/webpayments/issues/9#issuecomment-156973244">view it on GitHub</a>.<img=
 alt=3D"" height=3D"1" src=3D"https://github.com/notifications/beacon/APF=
rHM3wdMVEIRzVJu7g2_CeGulmAjDHks5pGZ42gaJpZM4GhP95.gif" width=3D"1" /></p>=

<div itemscope itemtype=3D"http://schema.org/EmailMessage">
<div itemprop=3D"action" itemscope itemtype=3D"http://schema.org/ViewActi=
on">
  <link itemprop=3D"url" href=3D"https://github.com/w3c/webpayments/issue=
s/9#issuecomment-156973244"></link>
  <meta itemprop=3D"name" content=3D"View Issue"></meta>
</div>
<meta itemprop=3D"description" content=3D"View this Issue on GitHub"></me=
ta>
</div>

----==_mimepart_5649a6b63281c_65343f8b4515b2a096388--

**Comment by msporny** [comment]
> - There is now a bot Github user (https://github.com/webpayments/) with
>   the email address github-w3c-webpayments-wg@w3.org

Correct.

> - That email account is subscribed to the WG mailing list and is also
>   subscribed to all Github issue notifications on the repo:
>   https://github.com/w3c/webpayments

It is also subscribed to:

https://github.com/WICG/web-payments-browser-api
https://github.com/WICG/paymentrequest
https://github.com/web-payments/web-payments-messaging/
https://github.com/web-payments/web-payments-http-api

and anything else the group deems appropriate as time goes on. Adding a new "watch" is as simple as logging in as the 'webpayments' user and "Watch"ing the repo (clicking a button).

> - When that account receives mails from Github it forwards them to the
>   mailing list.

Correct.

> - When the account receives replies from the mailing list it forwards
>   them to Github.

Technically, when people hit "Reply", they'll reply to the Github issue (via Github's mail server), which will then archive the email in the issue and send the response to the issue out to the mailing list.

Note that is is a feature that the standard W3C solution DOES NOT have. People w/o access to Github cannot participate in discussion.

> The result (once all moderation issues have been resolved) will be that:
> - If a new issue is logged or a comment is added to an existing issue
>   then a mail will be sent to the mailing list
> - Anyone replying to mails on the mailing list related to issues will
>   see their replies posted against the Github issue.

Correct.

> We still recommend that members of the WG post their comments and issues
> directly on Github if possible as this has the least chance of creating
> SPAM, messing with formatting or sending out duplicate messages.

Correct.

> Are there any outstanding security concerns:
> 1. Is it possible to compromise the Github account of the bot?

That would be highly unlikely.

We have two factor authentication turned on. So, an attacker would have to know the password on the account, steal one of the 'webpayments' admin's phones (Doug and myself right now, the chairs and W3C staff as soon as we can get you guys setup), and then figure out their password to unlock the phone.

> 1. Is it possible to post to Github "anonymously"?

No, but it's very easy to get a Github account using a fake email address and spam any issue on Github. It's also easy to subscribe to the mailing list and "Reply To" pseudo-anonymously. The easier attack is signing up for a Github account.

> 1. Are we certain that no email sent to the members-only mailing list
>    will end up on Github?

The members-only mailing list isn't setup at all to reflect issues to Github. The only way this could happen is if someone sent an email to members-only and cc'ed the public mailing list as well (which would kinda defeat the purpose of sending a members-only mail).

> Thanks for making this work. If it's as seamless as it appears it should be
> then I think it will be a great way to use Guthub and still accommodate
> those that like to (or have to) use the mailing list.

I think the solution meets all of the requirements we have and doesn't enable any kind of attack we can't control via the issue tracker. In the very worst case, we can shut down the spam and put a new (more restrictive) solution in place.

**Comment by webpayments** [comment]
*** NOTE: ***

    Your message was sent from an address which is not on the list
    of people who are authorized to post to this mailing list.
    Therefore, your message has been forwarded to the list maintainer
    for manual processing.

    If you do not see your message appear on the list or in the
    mailing list archives within a few business days, you may wish
    to contact the mailing list maintainer to investigate the delay.

    -- W3C Postmaster
       http://www.w3.org/Mail/

your message follows:
----------------------------------------------------------------------------


----==_mimepart_564a31676b924_45243ffcd09b72bc5794ae
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

>    - There is now a bot Github user (https://github.com/webpayments/) with
>    the email address github-w3c-webpayments-wg@w3.org

Correct.

>    - That email account is subscribed to the WG mailing list and is also
>    subscribed to all Github issue notifications on the repo:
>    https://github.com/w3c/webpayments

It is also subscribed to:

https://github.com/WICG/web-payments-browser-api
https://github.com/WICG/paymentrequest
https://github.com/web-payments/web-payments-messaging/
https://github.com/web-payments/web-payments-http-api

and anything else the group deems appropriate as time goes on. Adding a new "watch" is as simple as logging in as the 'webpayments' user and "Watch"ing the repo (clicking a button).

>    - When that account receives mails from Github it forwards them to the
>    mailing list.

Correct.

>    - When the account receives replies from the mailing list it forwards
>    them to Github.

Technically, when people hit "Reply", they'll reply to the Github issue (via Github's mail server), which will then archive the email in the issue and send the response to the issue out to the mailing list.

Note that is is a feature that the standard W3C solution DOES NOT have. People w/o access to Github cannot participate in discussion.

> The result (once all moderation issues have been resolved) will be that:
> 
>    - If a new issue is logged or a comment is added to an existing issue
>    then a mail will be sent to the mailing list
>    - Anyone replying to mails on the mailing list related to issues will
>    see their replies posted against the Github issue.

Correct.

> We still recommend that members of the WG post their comments and issues
> directly on Github if possible as this has the least chance of creating
> SPAM, messing with formatting or sending out duplicate messages.

Correct.

> Are there any outstanding security concerns:
> 
>    1. Is it possible to compromise the Github account of the bot?

That would be highly unlikely.

We have two factor authentication turned on. So, an attacker would have to know the password on the account, steal one of the 'webpayments' admin's phones (Doug and myself right now, the chairs and W3C staff as soon as we can get you guys setup), and then figure out their password to unlock the phone.

>    2. Is it possible to post to Github "anonymously"?

No, but it's very easy to get a Github account using a fake email address and spam any issue on Github. It's also easy to subscribe to the mailing list and "Reply To" pseudo-anonymously. The easier attack is signing up for a Github account.

>    3. Are we certain that no email sent to the members-only mailing list
>    will end up on Github?

The members-only mailing list isn't setup at all to reflect issues to Github. The only way this could happen is if someone sent an email to members-only and cc'ed the public mailing list as well (which would kinda defeat the purpose of sending a members-only mail).

> Thanks for making this work. If it's as seamless as it appears it should be
> then I think it will be a great way to use Guthub and still accommodate
> those that like to (or have to) use the mailing list.

I think the solution meets all of the requirements we have and doesn't enable any kind of attack we can't control via the issue tracker. In the very worst case, we can shut down the spam and put a new (more restrictive) solution in place.

---
Reply to this email directly or view it on GitHub:
https://github.com/w3c/webpayments/issues/9#issuecomment-157147357
----==_mimepart_564a31676b924_45243ffcd09b72bc5794ae
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<blockquote>
<ul>
<li>There is now a bot Github user (<a href="https://github.com/webpayments/">https://github.com/webpayments/</a>) with
the email address <a href="mailto:github-w3c-webpayments-wg@w3.org">github-w3c-webpayments-wg@w3.org</a>
</li>
</ul>
</blockquote>

<p>Correct.</p>

<blockquote>
<ul>
<li>That email account is subscribed to the WG mailing list and is also
subscribed to all Github issue notifications on the repo:
<a href="https://github.com/w3c/webpayments">https://github.com/w3c/webpayments</a>
</li>
</ul>
</blockquote>

<p>It is also subscribed to:</p>

<p><a href="https://github.com/WICG/web-payments-browser-api">https://github.com/WICG/web-payments-browser-api</a><br>
<a href="https://github.com/WICG/paymentrequest">https://github.com/WICG/paymentrequest</a><br>
<a href="https://github.com/web-payments/web-payments-messaging/">https://github.com/web-payments/web-payments-messaging/</a><br>
<a href="https://github.com/web-payments/web-payments-http-api">https://github.com/web-payments/web-payments-http-api</a></p>

<p>and anything else the group deems appropriate as time goes on. Adding a new "watch" is as simple as logging in as the 'webpayments' user and "Watch"ing the repo (clicking a button).</p>

<blockquote>
<ul>
<li>When that account receives mails from Github it forwards them to the
mailing list.</li>
</ul>
</blockquote>

<p>Correct.</p>

<blockquote>
<ul>
<li>When the account receives replies from the mailing list it forwards
them to Github.</li>
</ul>
</blockquote>

<p>Technically, when people hit "Reply", they'll reply to the Github issue (via Github's mail server), which will then archive the email in the issue and send the response to the issue out to the mailing list.</p>

<p>Note that is is a feature that the standard W3C solution DOES NOT have. People w/o access to Github cannot participate in discussion.</p>

<blockquote>
<p>The result (once all moderation issues have been resolved) will be that:</p>

<ul>
<li>If a new issue is logged or a comment is added to an existing issue
then a mail will be sent to the mailing list</li>
<li>Anyone replying to mails on the mailing list related to issues will
see their replies posted against the Github issue.</li>
</ul>
</blockquote>

<p>Correct.</p>

<blockquote>
<p>We still recommend that members of the WG post their comments and issues<br>
directly on Github if possible as this has the least chance of creating<br>
SPAM, messing with formatting or sending out duplicate messages.</p>
</blockquote>

<p>Correct.</p>

<blockquote>
<p>Are there any outstanding security concerns:</p>

<ol>
<li>Is it possible to compromise the Github account of the bot?</li>
</ol>
</blockquote>

<p>That would be highly unlikely.</p>

<p>We have two factor authentication turned on. So, an attacker would have to know the password on the account, steal one of the 'webpayments' admin's phones (Doug and myself right now, the chairs and W3C staff as soon as we can get you guys setup), and then figure out their password to unlock the phone.</p>

<blockquote>
<ol>
<li>Is it possible to post to Github "anonymously"?</li>
</ol>
</blockquote>

<p>No, but it's very easy to get a Github account using a fake email address and spam any issue on Github. It's also easy to subscribe to the mailing list and "Reply To" pseudo-anonymously. The easier attack is signing up for a Github account.</p>

<blockquote>
<ol>
<li>Are we certain that no email sent to the members-only mailing list
will end up on Github?</li>
</ol>
</blockquote>

<p>The members-only mailing list isn't setup at all to reflect issues to Github. The only way this could happen is if someone sent an email to members-only and cc'ed the public mailing list as well (which would kinda defeat the purpose of sending a members-only mail).</p>

<blockquote>
<p>Thanks for making this work. If it's as seamless as it appears it should be<br>
then I think it will be a great way to use Guthub and still accommodate<br>
those that like to (or have to) use the mailing list.</p>
</blockquote>

<p>I think the solution meets all of the requirements we have and doesn't enable any kind of attack we can't control via the issue tracker. In the very worst case, we can shut down the spam and put a new (more restrictive) solution in place.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/w3c/webpayments/issues/9#issuecomment-157147357">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/APFrHPv-aXfN5_qzvCw7nJTh4bTrucu4ks5pGijngaJpZM4GhP95.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/w3c/webpayments/issues/9#issuecomment-157147357"></link>
  <meta itemprop="name" content="View Issue"></meta>
</div>
<meta itemprop="description" content="View this Issue on GitHub"></meta>
</div>

----==_mimepart_564a31676b924_45243ffcd09b72bc5794ae--

**Comment by msporny** [comment]
Github to mailing list issues should now be resolved.

**Comment by msporny** [comment]
The cc'ed mailing list is now fixed, all emails should be cc'ed to public-payments-wg@w3.org.

**Comment by adrianhopebailie** [comment]
This appears to be working well.
Thanks @msporny
